PorousFlow: hystersis core doco added

WilkAndy opened the pull request:
Refs #16049

<!--
INCLUDE THE FOLLOWING IN THE PR DESCRIPTION
- Explain relevant design information for your change.
- Follow the [Coding Standards](http://mooseframework.org/wiki/CodeStandards/).
- Submit or improve [Test Cases](http://mooseframework.org/wiki/MooseTraining/testing/).
- Reference a specific issue, place "refs #<issue>" or "closes #<issue>" (e.g., #closes #1234).
-->

WilkAndy committed:
hystersis core doco added

Refs #16049

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/611150/) on 63162c5 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/16057/site)

This comment will be updated on new commits.

cpgr commented on modules/porous_flow/doc/content/bib/porous_flow.bib:
```diff
@@ -487,3 +487,46 @@ @book{javendel
+}
+
+@techreport{doughty2008,
+  title={User's guide for hysteretic capillary pressure and relative permeability functions in {iTOUGH2}, Rev3, Draft, March 2008},
```
The March renders with a lowercase m on the website

WilkAndy replied:
Thanks

cpgr commented on modules/porous_flow/doc/content/bib/porous_flow.bib:
```diff
@@ -487,3 +487,46 @@ @book{javendel
+author = {A. Niemi and G. S. Bodvarsson},
+year = {1988},
+pages = {277--291},
+title = {Preliminary capillary hysteresis simulations in fractured rocks, Yucca Mountain, Nevada},
```
Same with yucca mountain, nevada

WilkAndy replied:
done

cpgr commented on modules/porous_flow/doc/content/modules/porous_flow/hysteresis.md:
```diff
@@ -0,0 +1,171 @@
+# Hysteresis in PorousFlow
+
+Hysteretic capillary pressure and relative permeability functions are implemented in PorousFlow.  Hysteresis means that capillary pressure and relative permeability depend on the *history* as well as the current state.
```
Maybe the *saturation history*?

WilkAndy replied:
thanks, done

cpgr reviewed (commented):
Apart from these minor comments, thanks for the thorough documentation Andy

cpgr commented:
I'll add my comments for the other PR now too.

WilkAndy committed:
relperm info added.  Fixes #16049

WilkAndy commented:
Hey @cpgr, this is done now.  I addressed your comments above, and added stuff about relative permeability.  Could you read the relperm stuff?  (The cap pressure stuff is the same)

cpgr commented:
> Hey @cpgr, this is done now. I addressed your comments above, and added stuff about relative permeability. Could you read the relperm stuff? (The cap pressure stuff is the same)

yep. I'll take a look shortly.

WilkAndy commented:
Ooops, @cpgr - i just realised that i didn't push all the hysteresis relperm doco up yesterday.  i've just done that now and hopefully all the tests will pass - i'll send you the link for the doco when that happens.

WilkAndy commented:
The hysteresis doco, this time with the relperm stuff included, is at

https://mooseframework.inl.gov/docs/PRs/16057/site/modules/porous_flow/hysteresis.html

cpgr commented on modules/porous_flow/doc/content/modules/porous_flow/hysteresis.md:
```diff
@@ -0,0 +1,244 @@
+
+## Relative permeabilities
+
+Hysteresis is defined for two-phase systems only.  The water and gas relative permeability functions are both hysteretic.  Only the drying and first-order wetting curves are defined.  This means that if the system dries (following the drying curve) and then wets (following the first-order wetting curve) and then subsequently dries, the system will move along the first-order wetting curve until th ...
```
Typo (last word) - durve!

WilkAndy replied:
eagle eyes!

cpgr commented on modules/porous_flow/doc/content/modules/porous_flow/hysteresis.md:
```diff
@@ -0,0 +1,244 @@
+\end{aligned}
+\end{equation}
+
+Assuming the relative-permeability values are constant outside the well-defined domain results in curves of the form shown in [histeretic_krel_unextended_fig].
```
No hyphen?

cpgr replied:
In relative permeability, I mean.

cpgr commented on modules/porous_flow/doc/content/modules/porous_flow/hysteresis.md:
```diff
@@ -0,0 +1,244 @@
+   - for $S>S_{\mathrm{big}}$ the liquid drying curve, given in [relperm.eqns.0], is used
+   - otherwise, a cubic spline interpolates between these two, with parameters chosen so that the final result is continuous and has a continuous derivative.
+
+The result is illustrated in [histeretic_krel_extended_fig], using a cubic extension for the gas relative-permeability curve, and $r = 0.9$.
```
No hyphen?

cpgr commented on modules/porous_flow/doc/content/modules/porous_flow/hysteresis.md:
```diff
@@ -0,0 +1,244 @@
+
+The result is illustrated in [histeretic_krel_extended_fig], using a cubic extension for the gas relative-permeability curve, and $r = 0.9$.
+
+!media media/porous_flow/hysteretic_krel_extended.png caption=Extended and modified relative permeability curves, using a cubic extension for the gas relative-permeability curve, and $r$ being 0.9.  id=histeretic_krel_extended_fig
```
And here

cpgr commented on modules/porous_flow/doc/content/modules/porous_flow/hysteresis.md:
```diff
@@ -0,0 +1,244 @@
+
+### Examples
+
+!media media/porous_flow/hysteretic_krel_example_1.png caption=Example hysteretic relative-permeability functions.  The system initialises at full saturation, dries to saturation 0.5 (greater than $S_{l, r}$) and then wets back to saturation 1.  id=histeretic_krel_example_1
```
and here

cpgr commented on modules/porous_flow/doc/content/modules/porous_flow/hysteresis.md:
```diff
@@ -0,0 +1,244 @@
+
+!media media/porous_flow/hysteretic_krel_example_1.png caption=Example hysteretic relative-permeability functions.  The system initialises at full saturation, dries to saturation 0.5 (greater than $S_{l, r}$) and then wets back to saturation 1.  id=histeretic_krel_example_1
+
+!media media/porous_flow/hysteretic_krel_example_2.png caption=Example hysteretic relative-permeability functions.  The system initialises at full saturation, dries to saturation 0.1 (less than $S_{l, r}$) and then wets back to saturation 1.  In the extended region ($S_{l} \leq S_{l, r}$), the drying curve equals the wetting curve.  id=histeretic_krel_example_2
```
and finally here

cpgr reviewed (dismissed):
Looks great Andy.

WilkAndy committed:
@cpgr changes.  Fixes #16049

WilkAndy commented:
I believe this is ready for merging, @aeslaughter

WilkAndy commented:
Wow, that was quick, thanks @aeslaughter
